devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,210 @@
+			state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+		}
+		valueTile("tamper", "device.tamper", height: 2, width: 2, decoration: "flat") {
+			state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+			state "detected", label: 'tampered', backgroundColor: "#ffffff"
+		}
+
+		main "alarm"
+		details(["alarm", "off", "tamper"])
+	}
+}
+
+private getNumberOfSounds() {
+	def numberOfSounds = [
+			"0003": 8 //Aeotec Doorbell 6
+	]
+	return numberOfSounds[zwaveInfo.prod] ?: 1
+}
+
+def installed() {
+	initialize()
+	sendEvent(name: "alarm", value: "off", isStateChange: true, displayed: false)
+	sendEvent(name: "tamper", value: "clear", isStateChange: true, displayed: false)
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	if (!childDevices) {
+		addChildren(numberOfSounds)
+	}
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug "Parse returned: ${result.inspect()}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	createEvent(name: "alarm", value: cmd.value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def endpoint = cmd.sourceEndPoint
+	if (endpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def securedEncapsulatedCommand = cmd.securedEncapsulatedCommand()
+	if (securedEncapsulatedCommand) {
+		zwaveEvent(securedEncapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+private secure(physicalgraph.zwave.Command cmd) {
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private onOffCmd(value) {
+	secure(zwave.basicV1.basicSet(value: value))
+}
+
+def on() {
+	resetActiveSound()
+	state.lastTriggeredSound = 1
+	onOffCmd(0xFF)
+}
+
+def off() {
+	state.lastTriggeredSound = 1
+	onOffCmd(0x00)
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	secure(zwave.basicV1.basicGet())
+}
+
+private addChildren(numberOfSounds) {
+	for (def endpoint : 2..numberOfSounds) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+
+			addChildDevice("Aeotec Child Siren", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : "$device.displayName Sound $endpoint",
+					isComponent   : true,
+					componentName : "sound$endpoint",
+					componentLabel: "Sound $endpoint"
+			])
+		} catch (Exception e) {
+			log.debug "Excep: ${e} "
+		}
+	}
+}
+
+private encap(cmd, endpoint = null) {
```

Comment:
This command may require `secure` to be called as well?

Reply:
@MGoralczykS 👀

Reply:
@kianooshST Yes, I checked in docs and its required. I added this. 
@greens  sorry!

Reply:
Also I added here secure, but I didn't write this here.

Reply:
I believe @kianooshST was trying to get you to merge the `secure` and `encap` methods, so you could just call one everywhere and it would handle both encapsulations, one encapsulation, or none, depending on the arguments passed and device support. For example, right now you don't securely encapsulate the `childOn` and `childOff` commands.

Reply:
So as I understand this I should use secure on commands in lines: 157 and 159? Because we need to use secure for hubs with lower hubcore?

Reply:
Not exactly. I'm suggesting you merge the `encap` and `secure` methods and then just call that method everywhere instead of picking and choosing between encap and secure. It should be something like:
```
encap(cmd, endpoint = null)
    result = cmd
    if (endpoint)
        result = multichannelEncap(result)
    if (zwaveinfo.zw.contains("s"))
        result = secureEncap(result)
    return result
```

Reply:
Ok, could you check my last commit?